### PR TITLE
[Remove Stream/Optional: Phase 4] LPS-182656 Remove Stream and optional usages for modules: dynamic-data-mapping-data-provider-impl、dynamic-data-mapping-form-renderer

### PR DESCRIPTION
**Pull request by W2jx**
resend

Previous: https://github.com/liferay-core-infra/liferay-portal/pull/4382

### CODE REVIEW

**Comment by W2jx** [comment]
ci:test:forms

**Comment by Kvale1733** [comment]
@W2jx No regressions found.

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/rest/DDMRESTDataProvider.java [hunk 1]
```diff
@@ -523,18 +512,22 @@ private String _getPortalCacheKey(
...
 			ddmDataProviderId, StringPool.AT, url, StringPool.QUESTION,
-			stream.sorted(
-			).map(
-				keyValuePair -> StringBundler.concat(
-					keyValuePair.getKey(), StringPool.EQUAL,
-					keyValuePair.getValue())
-			).collect(
-				Collectors.joining(StringPool.AMPERSAND)
-			));
+			sb.toString());
```
Comment (dantewang) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-data-provider-impl/src/main/java/com/liferay/dynamic/data/mapping/data/provider/internal/rest/DDMRESTDataProvider.java:530]: @W2jx Please also combine this StringBundler.concat into the for-loop above.

**Review by dantewang — CHANGES_REQUESTED** [review]

**Comment by dantewang** [comment]
@W2jx Plz resend, thanks!

**Pull request closed without merging**